jina/executors/decorators.py

```diff
@@ -205,26 +205,28 @@ def _merge_results_after_batching(
     if not final_result:
         return
 
-    if isinstance(final_result[0], np.ndarray):
-        if len(final_result[0].shape) > 1:
+    if final_result and merge_over_axis is not None:
```

Comment:
the file has to be formatted with the pre-commit hook

Reply:
on it :rocket: